### PR TITLE
Fixing the TSLint task by removing deprecated rules.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: common/changes/ianc-removing-deprectated-tslint-rules_2016-12-22-19-50.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/gulp-core-build-typescript",
+      "comment": "Fixing TSLint task by removing some deprecated rules (\"label-undefined”, \"no-duplicate-key\", and \"no-unreachable\") and setting the \"noUnusedParameters\" and \"noUnusedLocals\" TS compiler options to cover the deprecated \"no-unused-variable\".",
+      "type": "patch"
```
Comment (cliffkoh) [inline common/changes/ianc-removing-deprectated-tslint-rules_2016-12-22-19-50.json]: The new compiler options are much stricter than no-unused-variable tslint (as can be seen by how this build is failing) so I think this should be considered a major/breaking change...

Comment (iclanton) [inline common/changes/ianc-removing-deprectated-tslint-rules_2016-12-22-19-50.json]: We already did the version bump of TSLint, so this is currently broken. It should be a patch on version 2.0.0, not 2.1.0 or 3.0.0.

Comment (cliffkoh) [inline common/changes/ianc-removing-deprectated-tslint-rules_2016-12-22-19-50.json]: Ok

Comment (cliffkoh) [inline common/changes/ianc-removing-deprectated-tslint-rules_2016-12-22-19-50.json]: The build is still broken though but I will approve assuming you'll fix the build break

**Pull request merged**